src/main/java/org/opensearch/ad/ratelimit/EntityFeatureRequest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.opensearch.ad.ratelimit;
+
+import org.opensearch.ad.model.Entity;
+
+public class EntityFeatureRequest extends EntityRequest {
+    private final double[] currentFeature;
+    private final long dataStartTimeMillis;
+
+    public EntityFeatureRequest(
+        long expirationEpochMs,
```

Comment:
how to decide this value? do we need this to be set per request?

Reply:
The expiry time is the start timestamp of the next detector run. Yes, it needs to be set per request.